test/katex-spec.js

```diff
@@ -2809,6 +2809,25 @@ describe("AMS environments", function() {
     });
 });
 
+describe("The CD environment", function() {
+    it("should fail if not is display mode", function() {
+        expect(`\\begin{CD}A @X<a<< B @>>b> C @>>> D\\\\@. @| @AcAA @VVdV \\\\@. E @= F @>>> G\\end{CD}`).not.toParse(
+            new Settings({displayMode: false})
+        );
+    });
+    const displaySettings = new Settings({displayMode: true});
+    it("should fail if the character after '@' is not in <>AV=|.", function() {
+        expect(`\\begin{CD}A @X<a<< B @>>b> C @>>> D\\\\@. @| @AcAA @VVdV \\\\@. E @= F @>>> G\\end{CD}`).not.toParse(displaySettings);
+    });
```

Comment:
It would be good to have a successfully test here as well.  It looks like the display mode test also has an issue with the `X` coming after `@`.  Having a successfully test case would help when determine which part of the other test cases are causing them not parse.

Reply:
Will do.